File: q6/q6.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if(numRows == 1){
            return s;
        }
        string solution;
        int size = s.size();
        int jump = 2*numRows-2;
        for (int i = 0; i < numRows; i++){
            for(int j = 0; j+i < size; j += jump){
                solution += s[j+i];
                if (i != 0 && i != numRows-1 && j+jump-i < size){
                    solution += s[j+jump-i];
                }
            }
        }
        return solution;
    }
};

File: q6/q6.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if(numRows == 1):
            return s;
        size = len(s)
        jump = 2*numRows -2
        strlist = []
        for i in range(numRows):
            for j in range(i, size, jump):
                strlist.append(s[j])
                if i != numRows -1  and i != 0 and j + jump - 2*i < size:
                    strlist.append(s[j + jump - 2*i])
        solution = ''.join(strlist)
        return solution
